packages/data-point/lib/entity-types/entity-schema/resolve.js

```diff
@@ -2,31 +2,38 @@ const _ = require('lodash')
 const Promise = require('bluebird')
 const Ajv = require('ajv')
 
+/**
+ * @param {Accumulator} acc
+ * @return {Promise}
+ */
 function validateContext (acc) {
   const ajv = new Ajv(acc.reducer.spec.options)
   const validate = ajv.compile(acc.reducer.spec.schema)
+  if (validate(acc.value)) {
+    return Promise.resolve(acc)
+  }
 
-  return Promise.resolve(validate(acc.value)).then(valid => {
-    if (!valid) {
-      const messages = _.map(validate.errors, 'message')
-      const messageListStr = messages.join('\n -')
-      const error = new Error(`Errors Found:\n - ${messageListStr}`)
-      error.name = 'InvalidSchema'
-      error.errors = validate.errors
-      return Promise.reject(error)
-    }
-
-    return acc
-  })
+  const message = _.map(validate.errors, 'message').join('\n -')
+  const error = new Error(`Errors Found:\n - ${message}`)
+  error.name = 'InvalidSchema'
+  error.errors = validate.errors
+  return Promise.reject(error)
```

Comment:
this just reformats the code to be a little easier to read (originally I had more changes to this function, but I ended up removing those changes)